metrics.go

```diff
@@ -423,18 +430,67 @@ func metricPointToMpbValue(pt *metricdata.Point) (*monitoringpb.TypedValue, erro
 				},
 			}
 		}
-		mv.DistributionValue.BucketCounts = addZeroBucketCountOnCondition(insertZeroBound, metricBucketToBucketCounts(dv.Buckets)...)
+		bucketCounts, exemplars := metricBucketToBucketCountsAndExemplars(dv.Buckets, projectID)
+		mv.DistributionValue.BucketCounts = addZeroBucketCountOnCondition(insertZeroBound, bucketCounts...)
+		mv.DistributionValue.Exemplars = exemplars
 
 		tval = &monitoringpb.TypedValue{Value: mv}
 	}
 
 	return tval, err
 }
 
-func metricBucketToBucketCounts(buckets []metricdata.Bucket) []int64 {
+func metricBucketToBucketCountsAndExemplars(buckets []metricdata.Bucket, projectID string) ([]int64, []*distributionpb.Distribution_Exemplar) {
 	bucketCounts := make([]int64, len(buckets))
+	var exemplars []*distributionpb.Distribution_Exemplar
 	for i, bucket := range buckets {
 		bucketCounts[i] = bucket.Count
+		if bucket.Exemplar != nil {
+			exemplars = append(exemplars, metricExemplarToPbExemplar(bucket.Exemplar, projectID))
+		}
+	}
+	return bucketCounts, exemplars
+}
+
+func metricExemplarToPbExemplar(exemplar *metricdata.Exemplar, projectID string) *distributionpb.Distribution_Exemplar {
+	return &distributionpb.Distribution_Exemplar{
+		Value:       exemplar.Value,
+		Timestamp:   timestampProto(exemplar.Timestamp),
+		Attachments: attachmentsToPbAttachments(exemplar.Attachments, projectID),
+	}
+}
+
+func attachmentsToPbAttachments(attachments metricdata.Attachments, projectID string) []*any.Any {
+	var pbAttachments []*any.Any
+	for _, v := range attachments {
+		switch v.(type) {
+		case string:
+			pbAttachments = append(pbAttachments, toPbStringAttachment(v))
```

Comment:
case string and default has same action. may be remove case string?

Reply:
SG, updated.